src/google-auth.js

```diff
@@ -275,3 +275,34 @@ GoogleSocialProvider.prototype.getAccessTokenFromRefreshToken_ =
     xhr.send({string: data});
   }.bind(this));
 };
+
+// returns Promise<accessToken>
+GoogleSocialProvider.prototype.refreshTokenIfNeeded_ = function(accessToken) {
+  // core.xhr callbacks clobber the xhr object if we use .bind(this), so
+  // instead we should set this to a local variable.
+  var thisSocialProvider = this;
+
+  return new Promise(function(fulfill, reject) {
+    var xhr = freedom["core.xhr"]();
+    var url = 'https://www.googleapis.com/oauth2/v1/tokeninfo?access_token=' +
```

Comment:
Just want to check if you should be doing something more like https://github.com/freedomjs/freedom-social-firebase/blob/dborkan-email-fix/src/google-social-provider.js#L32 (i.e. having the endPoint be a variable rather than hardcoded). It may well have to be hardcoded, but just want to be sure - thanks!